component/otelcol/host.go

```diff
@@ -0,0 +1,41 @@
+package otelcol
+
+import (
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelconfig "go.opentelemetry.io/collector/config"
+)
+
+// host implements otelcomponent.Host for Grafana Agent Flow.
+type host struct {
+	log log.Logger
+
+	extensions map[otelconfig.ComponentID]otelcomponent.Extension
+	exporters  map[otelconfig.DataType]map[otelconfig.ComponentID]otelcomponent.Exporter
```

Comment:
Thoughts on adding more structure to this? Maps in maps always feel off to me.

Reply:
🤷 It's the type OpenTelemetry requires for the Host interface. I can name the type but it still needs to be a map of maps.

Reply:
Fair enough.